builder-20/eol-buildpack/bin/build

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+function display_error() {
+  local ansi_red="\033[1;31m"
+  local ansi_reset="\033[0m"
+  local message=$1
+  echo
+  while IFS= read -r line; do
+    echo -e "${ansi_red}${line}${ansi_reset}" >&2
+  done <<< "$message"
+  echo
+}
+
+read -r -d '' EOL_MESSAGE <<'EOF' || true
```

Comment:
(optional) Upon David's [suggestion](https://github.com/heroku/cnb-builder-images/pull/429#issuecomment-1783192564), a later iteration of the EOL buildpack for the legacy shimmed builder images used this `display_error "$(cat <<EOF` pattern instead of `read -r`:
https://github.com/heroku/cnb-builder-images/blob/d74b3df909399e28e5ab39fad894ee86d3830a83/buildpacks-20/end-of-life-buildpack/bin/build#L32

Reply:
I just tried this - it failed pretty hard because of the special characters in the ascii art (like `(`, `)`, and `\`). I tried escaping them (which breaks legibility in the source), but that also fails, since `cat` doesn't respect the escapes.